Main.js

```diff
@@ -1,122 +1,237 @@
+            if(proposedXCoord < 1 || proposedXCoord > 7 || proposedYCoord < 1 || filledTiles.includes(proposedPosition)) {
+                console.log(`proposed position ${proposedPosition} is out of bounds or filled`);
+                return 'movement blocked';
             } else {
-                xCoordinate = newXCoord;
-            } 
-        break;
-        default:
-            console.log(`input ${movementInput} ignored`);
-        break;
+                updateTetraminoPosition(proposedPosition);
+                return proposedPosition;
+            }
+
+        } else {
+            console.log(`input "${movementInput}" ignored`);
+        }
+
+    } else {
+        console.log(`movement disallowed, input ignored`);
     }
-    let newPosition = xCoordinate.toString() + yCoordinate.toString();
-    console.log(`new x and y coords are ${xCoordinate} and ${yCoordinate} newPosition is ${newPosition}`)
-    document.getElementById(newPosition).style.backgroundColor = '#A9A9A9';
-    tetraminoPosition = newPosition;
-    document.getElementById('tetraminoPos').innerHTML = tetraminoPosition;
 }
 
-var fallingTetra = null;
 
-function dropTetra() {
-    let oldPosition = tetraminoPosition;
-    let newPosition = oldPosition[0] + (Number(oldPosition[1])-1).toString();
-    console.log(`old and new positions are ${oldPosition} and ${newPosition}`);
-    if (tetraminoPosition[1] === '1' || filledTiles.includes(newPosition)) {
-        filledTiles.push(oldPosition);
-        tetraminoPosition = '47';
-        document.getElementById(tetraminoPosition).style.backgroundColor = '#A9A9A9';
-    } else {
-        moveTetra('s');
+//function to move the tetra down, filling the position if the movement fails (function returns 'movement blocked')
+//ends the game if the tile stops at position 49
+function dropTile() {
+    let newPosition = moveTile('s');
+    if(newPosition === 'movement blocked'){
+        console.log(`movement blocked, position ${tetraminoPosition} is now filled`);
+        filledTiles.push(tetraminoPosition);
+        tetraminoPosition = '49';
+        document.getElementById('49').style.backgroundColor = '#A9A9A9';
+    }
+    score();
+    if(filledTiles.includes('49')){
+        movementAllowed = false;
+        fallingState(false);
+        document.getElementById('gameOverScreen').style.display = 'block';
     }
-    document.getElementById('tetraminoPos').innerHTML = tetraminoPosition;
 }
 
+//function to start/stop the tile falling
+//uses a global variable to store the state
+let fallingTetra;
 function fallingState(inputBool) {
     if(inputBool) {
-        fallingTetra = setInterval(dropTetra, 500);
+        fallingTetra = setInterval(dropTile, 500);
     } else {
         clearInterval(fallingTetra);
     }
 }
 
-fallingState(true);
+//function resets tile position to 49, clears all the tile backgrounds back to white, hides the game over / game title screen and starts the tile falling
```

Comment:
getting a null error here, must check generatedTiles array. Possibly has something to do with adding a classname to the tiles